app/services/course/material/zip_download_service.rb

```diff
@@ -5,8 +5,51 @@ class << self
     # @param [Course::Material::Folder] folder The folder containing the materials.
     # @param [Array<Course::Material>] materials The materials to be downloaded.
     # @return [String] The path to the zip file.
-    def download_and_zip(_folder, _materials)
-      # TODO: To implement
+    def download_and_zip(folder, materials)
+      service = Course::Material::ZipDownloadService.new(folder, materials)
+      service.download_to_base_dir
+      service.zip_base_dir
     end
   end
+
+  # Downloads the materials to the the base directory.
+  def download_to_base_dir
```

Comment:
Okay, then no choice, make the instance methods private, use send in the class method...